src/main/java/com/atlassian/jira/cloud/jenkins/util/XmlUtils.java

```diff
@@ -0,0 +1,49 @@
+package com.atlassian.jira.cloud.jenkins.util;
+
+import hudson.XmlFile;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Utility class for handling XML-related operations such as parsing XML files and extracting values
+ * from XML elements.
+ *
+ * <p>This class provides methods to parse XML files, extract specific values based on tag names,
+ * and handle boolean values stored within XML elements.
+ */
+public class XmlUtils {
+
+    public Element parseXmlFile(final XmlFile file)
```

Comment:
Should we have a try catch in this bad boy? What is xml parsing fails? Would be good to have something to capture those exceptions.

Reply:
have wrapped it up, but then rethrow :| the calling function logic relies on the exception being thrown as opposed to returning null etc, the change does make it easier to pinpoint the issue tho